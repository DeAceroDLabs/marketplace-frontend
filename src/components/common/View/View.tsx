import React from "react";
import styles from "./View.module.scss";

interface ViewProps {
  header?: React.ReactNode;
  body?: React.ReactNode;
  footer?: React.ReactNode;
}

const View: React.FC<ViewProps> = ({ header, body, footer }) => (
  <>
    <div className={styles.header}>
      <div>{header}</div>
    </div>
    <div className={styles.body}>
      <div>{body}</div>
    </div>
    <div className={styles.footer}>
      <div>{footer}</div>
    </div>
  </>
);

export default View;
