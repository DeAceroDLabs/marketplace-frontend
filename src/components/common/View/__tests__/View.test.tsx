import renderer from "react-test-renderer";
import View from "../View";

it("renders View with no issue", () => {
  const component = renderer.create(<View />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
