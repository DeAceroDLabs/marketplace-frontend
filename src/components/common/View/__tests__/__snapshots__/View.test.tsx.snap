// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders View with no issue 1`] = `
Array [
  <div
    className="header"
  >
    <div />
  </div>,
  <div
    className="body"
  >
    <div />
  </div>,
  <div
    className="footer"
  >
    <div />
  </div>,
]
`;
